engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -568,6 +568,64 @@ yield switch (op.codeRepr()) {
             );
             yield prefix;
           }
+          case "->" -> {
+            if (insideTypeSignature) {
+              List<IR.Expression> args = nil();
+              Tree treeBody = null;
+              Tree.OprApp head = app;
+              while (head != null) {
+                // FIXME: Handle parameterized types here.
```

Comment:
Do we have an Enso source example that needs handling?

Reply:
Added a test case, with `@Ignore` for now. Before this PR, the test case fails due to incorrect IR (the previous code here flattened all types and their parameters into one list); after this PR, the unhandled case throws an exception from the line under this comment.